File: for-local-test/205.isomorphic-strings.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> sINt = new HashMap<>();
        Map<Character, Character> tINs = new HashMap<>();
        for (int i = 0; i < s.length(); i++){
            char sChar = s.charAt(i);
            char tChar = t.charAt(i);
            if(sINt.containsKey(sChar) && sINt.get(sChar) != tChar) return false;
            if(tINs.containsKey(tChar) && tINs.get(tChar) != sChar) return false;
            sINt.putIfAbsent(sChar, tChar);
            tINs.putIfAbsent(tChar, sChar);
        }
        return true;
    }
    public static void main(String[] args) {
        Solution solution = new Solution();
        System.out.println(solution.isIsomorphic("egg", "add")); // true
        System.out.println(solution.isIsomorphic("foo", "bar")); // false
        System.out.println(solution.isIsomorphic("paper", "title")); // true
    }
}

File: for-local-test/python/205.isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        s_to_t = {}
        t_to_s = {}
        for cs, ct in zip(s, t):
            if(cs in s_to_t and s_to_t[cs] != ct):
                return False
            if(ct in t_to_s and t_to_s[ct] != cs):
                return False
            s_to_t[cs] = ct
            t_to_s[ct] = cs
        return True
    
if __name__ == "__main__":
    solution = Solution()
    print(solution.isIsomorphic("egg", "add")) # true
    print(solution.isIsomorphic("foo", "bar")) # false
    print(solution.isIsomorphic("paper", "title")) # true